src/components/withList.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+const withList = (ListItemComponent, propName = 'list') => {
```

Comment:
nice use of withList- overall really clean, well organized code all around. Great job!